### ISSUE TITLE
Clickhouse dialect does not understand CTAS query

### ISSUE THREAD

**Issue by was-av** [sqlfluff/sqlfluff#3860]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Lint query with CTAS query and got lint error 'unparsable section'.

### Expected Behaviour

Successfull liniing

### Observed Behaviour

Clickhouse dialect should support CTAS queries.

### How to reproduce

1. Create file with query

```sql
-- query.sql
create table t1 engine = Log() as SELECT 1 as id;
```
2. Lint query by sqlfluff
```shell
$ sqlfluff lint query.sql --dialect clickhouse

== [test_ch_ctas.sql] FAIL                                                          
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'create
                       | table t1 engine = Log() as SELECT...'
```

### Dialect

clickhouse

### Version

Python 3.10.6
sqlfluff, version 1.3.1

### Configuration

[sqlfluff]
dialect = clickhouse

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Create Table As statements are supported generally. This statement parses fine: `create table t1 as SELECT 1 as id;`.

The part here that doesn't parse is the `engine = Log()` syntax which I'm not familiar with. If that's a clickhouse specific feature of that dialect it will need to be added to the clickhouse dialect. Do you have the documentation of the syntax for that feature of the dialect?

**Comment by was-av** [comment]
Of course.
https://clickhouse.com/docs/en/sql-reference/statements/create/table#from-select-query

**Comment by was-av** [comment]
Reusing logic from https://github.com/sqlfluff/sqlfluff/pull/3986/files can help implement this feature.
@alanmcruickshank

**Comment by Ilya1991** [comment]
Facing the same issue.
+1

**Comment by konnectr** [comment]
This issue works on 2.0.0a5 
There is even a test 
https://github.com/sqlfluff/sqlfluff/blob/main/test/fixtures/dialects/clickhouse/create_table.sql#L36
